R/messages.R

```diff
@@ -1,84 +1,245 @@
+  return(x)
 }
 
-build_function_not_called_msg <- function(name, index) {
-  lang <- get_language()
-  if (lang == "en") {
-    msg <- sprintf("The system wants to check the %s call of `%s()`, but it hasn't found it; have another look at your code.", 
-                   get_num(index), name)
-  } else if (lang == "fr") {
-    msg <- sprintf("Avez-vous ex&#233;cut&#233; %d fois la fonction `%s()` ?", 
-                   index, name)
-  } else if (lang == "es") {
-    msg <- sprintf("&#191;Usaste la funci&#243;n `%s()` %s?", 
-                   name, ifelse(index == 1, "una vez", sprintf("%d veces", index)))
-  } else {
-    stop(no_msg)
-  }
-  return(msg)
+`%+=%` <- function(a, b) {
+  eval.parent(substitute(a <- paste(a, b)))
 }
 
-build_function_args_not_specified_msg <- function(name, args, n_args, allow_extra) {
-  lang <- get_language()
-  if(lang == "en") {
-    msg <- sprintf("Did you specify the argument%s %s in your call of `%s()`?%s",
-                   ifelse(n_args == 1, "", "s"),
-                   collapse_args(args),
-                   name,
-                   ifelse(allow_extra, "", " You shouldn't specify any other arguments!"))
-  } else if (lang == "fr") {
-    msg <- sprintf("Avez-vous specifi&#233; %sargument%s %s dans la fonction `%s()` ?%s",
-                   ifelse(n_args == 1, "l'", "les "),
-                   ifelse(n_args == 1, "", "s"),
-                   collapse_args(args, " et "),
-                   name,
-                   ifelse(allow_extra, "", " Ne specifiez pas d'autres arguments !"))
-  } else if (lang == "es") {
-    msg <- sprintf("Especifcaste %s argumento%s %s en la funci&#243;n `%s()`? %s",
-                   ifelse(n_args == 1, "el", "los"),
-                   ifelse(n_args == 1, "", "s"),
-                   collapse_args(args, " y "),
-                   name,
-                   ifelse(allow_extra, "", " No especifica otros argumentos!"))
-  } else {
-    stop(no_msg)
-  }
-  return(msg)
+`%+=0%` <- function(a, b) {
+  eval.parent(substitute(a <- paste0(a, b)))
 }
 
-build_function_incorrect_msg <- function(name, incorrect_arg) {
-  lang <- get_language()
-  if (lang == "en") {
-    msg <- sprintf("Did you correctly specify the argument `%s` in your call of `%s()`?", 
-                   incorrect_arg, name)
-  } else if (lang == "fr") {
-    msg <- sprintf("Avez-vous affect&#233; la bonne valeur &#224; l'argument `%s` dans la fonction `%s()` ?",
-                   incorrect_arg, name)
-  } else if (lang == "es") {
-    msg <- sprintf("&#191;Usaste el valor correcto para el argumento `%s` en la funci&#243;n `%s()`?",
-                   incorrect_arg, name)
-  } else {
-    stop(no_msg)
+build_feedback_message <- function(details) {
+  if (is.character(details)) {
+    return(capitalize(trim(details)))
   }
-  return(msg)
   
+  total_msg <- ""
+  for (det in details) {
+    if (!is.null(det$message)) {
+      msg <- det$message
+    } else {
+      msg <- ""
```

Comment:
Couldn't you avoid this giant if statement by using an S3 function:
```
class(det) <- det$type
msg <- generate_message(det)
```
You could use S3 functions to split further using `det$case` as well, or just use `switch(...)` statements there?